From ae214228e0dbe2cb6428fd9185be4a5e9e61c6b6 Mon Sep 17 00:00:00 2001
From: nxf56319 <sai.shwethas@nxp.com>
Date: Wed, 28 Aug 2024 15:31:50 +0530
Subject: [PATCH 09/18] [artf1176907]: Fix for NFC service die during first
 boot

1. Remove phNxpNciHal_deinitializeRegRfFwDnld as part of factory reset.
2. fpDoAntennaActivity should be set to NULL as libonebinary is closed.

Change-Id: I7d85ef1d728d1969022fd10b241f7238743018df
---
 snxxx/halimpl/hal/phNxpNciHal.cc | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/snxxx/halimpl/hal/phNxpNciHal.cc b/snxxx/halimpl/hal/phNxpNciHal.cc
index 904c5e00..d4037e2d 100644
--- a/snxxx/halimpl/hal/phNxpNciHal.cc
+++ b/snxxx/halimpl/hal/phNxpNciHal.cc
@@ -3778,19 +3778,23 @@ static NFCSTATUS phNxpNciHal_do_swp_session_reset(void) {
  ******************************************************************************/
 void phNxpNciHal_do_factory_reset(void) {
   NFCSTATUS status = NFCSTATUS_FAILED;
+  bool isHalOpenRequested = false;
   // After factory reset phone will turnoff so mutex not required here.
   if (nxpncihal_ctrl.halStatus == HAL_STATUS_CLOSE) {
+    isHalOpenRequested = true;
     status = phNxpNciHal_MinOpen();
     if (status != NFCSTATUS_SUCCESS) {
       NXPLOG_NCIHAL_E("%s: NXP Nfc Open failed", __func__);
       return;
     }
-    phNxpNciHal_deinitializeRegRfFwDnld();
   }
   status = phNxpNciHal_do_swp_session_reset();
   if (status != NFCSTATUS_SUCCESS) {
     NXPLOG_NCIHAL_E("%s failed. status = %x ", __func__, status);
   }
+  if (nxpncihal_ctrl.halStatus == HAL_STATUS_MIN_OPEN && isHalOpenRequested) {
+    phNxpNciHal_close(false);
+  }
 }
 /******************************************************************************
  * Function         phNxpNciHal_hci_network_reset
@@ -4136,6 +4140,7 @@ void phNxpNciHal_deinitializeRegRfFwDnld() {
     fpPropConfCover = NULL;
     dlclose(RfFwRegionDnld_handle);
     RfFwRegionDnld_handle = NULL;
+    fpDoAntennaActivity = NULL;
   }
 }
 
-- 
2.47.0

